oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -182,16 +182,15 @@ public OSProcess getProcess(int pid) {
             }
         } else {
             // Only put descendants in
```

Comment:
Your changes work for this but looking at the whole method there's opportunity for more improvement.  You could make this branch the top one (`if (parentPid >= 0)`) and just do `return queryProcessList(getChildrenOrDescendants...)`. With the return no need for `else`.  Then handle the other (`-1`) case by declaring the set and adding to it as is currently done.  I see a bug, we shouldn't add the default `0` to the list of children.  Can you fix that please? :)

Reply:
My first instinct to fix the default 0 bug was to replace it with `Integer.parseInt()` but from seeing your other reply I'm guessing this isn't correct because it could throw an exception. Should I be changing it to `ParseUtil.parseIntOrDefault(procFile.getName(), -1)` instead?

Reply:
Or actually should I just add an if statement to only add it if it's not 0?

Reply:
In some OS's (Windows) 0 is a valid PID.  In *nix, I think 1 is the minimum, so 0 could be safe here in later code but still shouldn't be added.  Any negative number should also work*, but note that a few methods use -1 as the input PID to mean "get all the PIDs" so you'll have to be careful.

But as you just said, if you separate the retrieving of the value and an if statement adding it to the set, then you can use any value.

* Actually if windows has a process ID higher than Integer.MAX_VALUE it would show up as negative!  But this is Linux so that doesn't matter.

Reply:
Thanks for all the information and help! Last question (for now at least), when I'm finished with my changes am I supposed amend and push my first commit or push a new commit?

Reply:
Either is fine.  GitHub has a Squash and merge” option for me to use.  

Some other projects prefer you squash and set your real name and email as well.  I’m not that picky. :)